### PR TITLE
Fix more types (8)

### PR DESCRIPTION
**Pull request by hawkrives**
Clean up most "unexpected `any`" warnings. 

Make Markdown module complain a lot less (needs testing in simulator!)

Clean up a bunch of warnings in the Tableview module. 

Fixed a problem in HomeScreenNotice. 

Fixed some issues in FAQs. 

Clean up some lint warnings in ContactsDetail.

### CODE REVIEW

File: modules/markdown/markdown.tsx [hunk 1]
```diff
@@ -1,68 +1,101 @@
...
+		Emph?: StyleProp<TextStyle>
+		Heading?: StyleProp<TextStyle>
+		Image?: StyleProp<ViewStyle>
+		ListItem?: StyleProp<TextStyle>
+		Link?: StyleProp<TextStyle>
+		List?: StyleProp<ViewStyle>
+		Paragraph?: StyleProp<TextStyle>
+		Strong?: StyleProp<TextStyle>
+		ThematicBreak?: StyleProp<TextStyle>
+	}
```
Comment (drewvolz) [inline modules/markdown/markdown.tsx:47]: It is super nice to have these styles types with their styleprops!

File: modules/markdown/markdown.tsx [hunk 2]
```diff
@@ -1,68 +1,101 @@
...
 }
 
 export class Markdown extends React.PureComponent<Props> {
-	render() {
+	render(): JSX.Element {
 		const {styles = {}, source} = this.props
 		return (
 			<ReactMarkdown
-				containerTagName={View}
+				containerTagName={View as unknown as string}
```
Comment (drewvolz) [inline modules/markdown/markdown.tsx:56]: Great to see the issue with upgrading (previously blocked by types) is unblocked by this.

File: modules/tableview/cells/delete-button.tsx [hunk 3]
```diff
@@ -14,14 +15,14 @@ export const DeleteButtonCell = ({
 }: {
 	title: string
 	skipConfirm?: boolean
-	onPress?: () => any
+	onPress?: () => void
 }): JSX.Element => {
-	let onPressCallback = onPress ? onPress : () => {}
+	let onPressCallback = onPress ?? noop
```
Comment (drewvolz) [inline modules/tableview/cells/delete-button.tsx:20]: Nice, I like this use of noop here.

**Review by drewvolz — APPROVED** [review]
This is great!

**Review by rye — APPROVED** [review]

File: modules/markdown/markdown.tsx (hunk 2 above)
Comment (hawkrives) [reply modules/markdown/markdown.tsx:56]: "This is a hack"

I'll have to try upgrading to react-markdown@8 again, and see how it goes. (In another PR)

Worst case, I write a custom markdown-dom-to-React component for us to use. (I don't need a custom parser, just a custom transformation on top of the parsed output.)

**Pull request merged**